src/test/java/it/gov/pagopa/reward/service/synchronous/RewardTrxSynchronousApiServiceImplTest.java

```diff
@@ -0,0 +1,168 @@
+
+        Mockito.when(rewardContextHolderServiceMock.getInitiativeConfig(initiativeId)).thenReturn(Mono.empty());
+
+        SynchronousTransactionResponseDTO response = SynchronousTransactionResponseDTOFaker.mockInstanceBuilder(1)
+                .initiativeId(initiativeId)
+                .status(RewardConstants.REWARD_STATE_REJECTED)
+                .rejectionReasons(List.of(RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE))
+                .build();
+        Mockito.when(synchronousTransactionRequestDTOt2TrxDtoOrResponseMapperMock.apply(previewRequest, initiativeId, List.of(RewardConstants.TRX_REJECTION_REASON_INITIATIVE_NOT_FOUND))).thenReturn(response);
+
+        RewardTrxSynchronousApiServiceImpl rewardTrxSynchronousApiApiService = new RewardTrxSynchronousApiServiceImpl(rewardContextHolderServiceMock, onboardedInitiativesServiceMock,initiativesEvaluatorFacadeServiceMock, userInitiativeCountersRepositoryMock, synchronousTransactionRequestDTOt2TrxDtoOrResponseMapperMock, rewardTransaction2SynchronousTransactionResponseDTOMapperMock);
+
+        // When
+        try {
+            rewardTrxSynchronousApiApiService.previewTransaction(previewRequest, initiativeId).block();
+
+        } catch (Exception e){
+            Assertions.assertTrue(e instanceof TransactionSynchronousException);
+            SynchronousTransactionResponseDTO resultResponse = ((TransactionSynchronousException) e).getResponse();
+            Assertions.assertNotNull(resultResponse);
+            SynchronousTransactionRequestDTOt2TrxDtoOrResponseMapperTest.errorResponseCommonAssertions(previewRequest, initiativeId, List.of(RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE), resultResponse);
+
+        }
+    }
+
+    @Test
+    void postTransactionPreviewOK(){
+        // Given
+        SynchronousTransactionRequestDTO previewRequest = SynchronousTransactionRequestDTOFaker.mockInstance(1);
+        String initiativeId = "INITIATIVEID";
+
+        TransactionDTO transactionDTOMock = TransactionDTOFaker.mockInstance(1);
+        InitiativeConfig initiativeConfig = InitiativeConfig.builder()
+                .initiativeId(initiativeId)
+                .build();
+        Mockito.when(rewardContextHolderServiceMock.getInitiativeConfig(initiativeId)).thenReturn(Mono.just(initiativeConfig));
+
+        Mockito.when(onboardedInitiativesServiceMock.isOnboarded(Mockito.same(transactionDTOMock.getHpan()), Mockito.same(transactionDTOMock.getTrxDate()), Mockito.same(initiativeId))).thenReturn(Mono.just(Boolean.TRUE));
+
+        Mockito.when(synchronousTransactionRequestDTOt2TrxDtoOrResponseMapperMock.apply(Mockito.same(previewRequest))).thenReturn(transactionDTOMock);
+
+        UserInitiativeCounters userInitiativeCounters = new UserInitiativeCounters();
+        userInitiativeCounters.setUserId(transactionDTOMock.getUserId());
+        Mockito.when(userInitiativeCountersRepositoryMock.findById(Mockito.anyString())).thenReturn(Mono.just(userInitiativeCounters));
```

Comment:
perform the same test, but returning empty here (these tests will differ just for this line, you could let the thenReturn configuration a parameter of a common method invoked by both tests